plugins/redis/redis.cpp

```diff
@@ -408,37 +745,113 @@ Connection * Connection::createConnection(ICodeContext * ctx,  Connection * & _c
             connectionCachingLevelChecked = true;
         }
     }
+    return connectionCachingLevel;
+}
+bool Connection::getCacheSubConnections()
+{
+    if (!cacheSubConnectionsOptChecked)//cacheSubConnectionsOptChecked is std:atomic<bool>. Test to guard against unnecessary critical section
+    {
+        CriticalBlock block(critsec);
+        if (!cacheSubConnectionsOptChecked)
+        {
+            const char * tmp = getenv("HPCC_REDIS_PLUGIN_CACHE_SUB_CONNECTIONS");
+            //cacheSubConnections is already defaulted to true;
+            if (tmp && *tmp)
+            {
+                StringBuffer buf(tmp);
+                buf.toLowerCase();
+                //less ops to check on & true however, making cacheSubConnections(false) if not met would do also for misspelled versions.
+                //Since the default is ON, misspelling on or true means/does nothing.
+                if (*tmp == '0' || (strcmp("off", buf.str()) == 0) || (strcmp("false", buf.str()) == 0))
```

Comment:
minor:  This code is fine, but you could have used stricmp() and avoided the call to lower case the string.